backend/plugin/db/risingwave/risingwave.go

```diff
@@ -0,0 +1,458 @@
+		dbBinDir: config.DbBinDir,
+	}
+}
+
+// Open opens a RisingWave driver.
+func (driver *Driver) Open(_ context.Context, _ db.Type, config db.ConnectionConfig, _ db.ConnectionContext) (db.Driver, error) {
+	// Require username for Postgres, as the guessDSN 1st guess is to use the username as the connecting database
+	// if database name is not explicitly specified.
+	if config.Username == "" {
+		return nil, errors.Errorf("user must be set")
+	}
+
+	if config.Host == "" {
+		return nil, errors.Errorf("host must be set")
+	}
+
+	if config.Port == "" {
+		return nil, errors.Errorf("port must be set")
+	}
+
+	if (config.TLSConfig.SslCert == "" && config.TLSConfig.SslKey != "") ||
+		(config.TLSConfig.SslCert != "" && config.TLSConfig.SslKey == "") {
+		return nil, errors.Errorf("ssl-cert and ssl-key must be both set or unset")
+	}
+
+	connStr := fmt.Sprintf("host=%s port=%s", config.Host, config.Port)
+	connConfig, err := pgx.ParseConfig(connStr)
+	if err != nil {
+		return nil, err
+	}
+	connConfig.Config.User = config.Username
+	connConfig.Config.Password = config.Password
+	connConfig.Config.Database = config.Database
+	if config.TLSConfig.SslCert != "" {
+		cfg, err := config.TLSConfig.GetSslConfig()
+		if err != nil {
+			return nil, err
+		}
+		connConfig.TLSConfig = cfg
+	}
+	if config.SSHConfig.Host != "" {
+		sshClient, err := util.GetSSHClient(config.SSHConfig)
+		if err != nil {
+			return nil, err
+		}
+		driver.sshClient = sshClient
+
+		connConfig.Config.DialFunc = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			conn, err := sshClient.Dial(network, addr)
+			if err != nil {
+				return nil, err
+			}
+			return &noDeadlineConn{Conn: conn}, nil
+		}
+	}
+	if config.ReadOnly {
+		connConfig.RuntimeParams["default_transaction_read_only"] = "true"
+	}
+
+	driver.databaseName = config.Database
+	if config.Database == "" {
+		databaseName, cfg, err := guessDSN(connConfig, config.Username)
+		if err != nil {
+			return nil, err
+		}
+		connConfig = cfg
+		driver.databaseName = databaseName
+	}
+	driver.config = config
+
+	driver.connectionString = stdlib.RegisterConnConfig(connConfig)
+	db, err := sql.Open(driverName, driver.connectionString)
+	if err != nil {
+		return nil, err
+	}
+	driver.db = db
+	return driver, nil
+}
+
+type noDeadlineConn struct{ net.Conn }
+
+func (*noDeadlineConn) SetDeadline(time.Time) error      { return nil }
+func (*noDeadlineConn) SetReadDeadline(time.Time) error  { return nil }
+func (*noDeadlineConn) SetWriteDeadline(time.Time) error { return nil }
+
+// guessDSN will guess a valid DB connection and its database name.
+func guessDSN(baseConnConfig *pgx.ConnConfig, username string) (string, *pgx.ConnConfig, error) {
+	// Some postgres server default behavior is to use username as the database name if not specified,
+	// while some postgres server explicitly requires the database name to be present (e.g. render.com).
+	guesses := []string{"postgres", username, "template1"}
```

Comment:
RisingWave seems do not contain `postgres` and `template1` database, should we use `dev` database in guessing?